New ndimage and RBFInterpolator test failures in pre-release CI job

See for example [this CI log](https://github.com/scipy/scipy/actions/runs/7493606138/job/20400251179?pr=19866). There's 101 failures, 3 for `test_measurements.py`  like the one below, and 98 from `pythran` in `RBFInterpolator`:
```
FAILED scipy/ndimage/tests/test_measurements.py::Test_measurements_stats::test_nonint_labels - ValueError: object too deep for desired array
FAILED scipy/interpolate/tests/test_rbfinterp.py::TestRBFInterpolatorNeighbors20::test_scale_invariance_1d[cubic] - TypeError: only integer scalar arrays can be converted to a scalar index
```

This seems to have started roughly 24 hours ago, and seems unrelated to other recent failures. It is the `Meson build (3.12, false)` job rather than the one labeled pre-release, but looking at the job it does install `pythran` master and `meson` master (because it used to support 3.12-dev).

The Pythran errors are:
```
E   TypeError: only integer scalar arrays can be converted to a scalar index
```
and the ndimage ones are:
```
scipy/ndimage/_measurements.py:660: in _stats
    counts = numpy.bincount(new_labels)
E   ValueError: object too deep for desired array
```
both of which is a little puzzling. However, I can reproduce them with numpy's `main` branch - something changed there.

--- a/scipy/interpolate/_rbfinterp.py
+++ b/scipy/interpolate/_rbfinterp.py
@@ -511,6 +511,7 @@ class RBFInterpolator:
             # neighborhood.
             yindices = np.sort(yindices, axis=1)
             yindices, inv = np.unique(yindices, return_inverse=True, axis=0)
+            inv = np.reshape(inv, (-1,))  # flatten, we need 1-D indices
             # `inv` tells us which neighborhood will be used by each evaluation
             # point. Now we find which evaluation points will be using each
             # neighborhood.

--- a/scipy/ndimage/_measurements.py
+++ b/scipy/ndimage/_measurements.py
@@ -657,6 +657,7 @@ def _stats(input, labels=None, index=None, centered=False):
         # be 1-D, but it should be interpreted as the flattened N-D array of
         # label indices.
         unique_labels, new_labels = numpy.unique(labels, return_inverse=True)
+        new_labels = np.reshape(new_labels, (-1,))  # flatten, since it may be >1-D
         counts = numpy.bincount(new_labels)
         sums = numpy.bincount(new_labels, weights=input.ravel())
         if centered:

